world_boss/app/raid.py

```diff
@@ -262,3 +268,235 @@ def get_jwt_auth_header() -> dict[str, str]:
+    # avatar_address : world_boss_reward
+    world_boss_rewards: dict[str, dict] = {}
+    signer_address = signer.address
+    tx_values: List[dict] = []
+    tx_ids: dict[int, str] = {}
+    for n in nonce_rows_map.keys():
+        recipient_rows = nonce_rows_map[n]
+        recipients = [row_to_recipient(r) for r in recipient_rows]
+        pv = get_transfer_assets_plain_value(signer_address, recipients, memo)
+        unsigned_transaction = create_unsigned_tx(
+            config.planet_id, signer.public_key, signer_address, n, pv, time_stamp
+        )
+        signed_transaction = signer.sign(unsigned_transaction, n)
+        tx_id = hashlib.sha256(signed_transaction).hexdigest()
+        tx_values.append(
+            {
+                "tx_id": tx_id,
+                "nonce": n,
+                "signer": signer_address,
+                "payload": signed_transaction.hex(),
+                "tx_result": "CREATED",
+            }
+        )
+        tx_ids[n] = tx_id
+    db.execute(insert(Transaction), tx_values)
+    raid_id = int(rows[0][0])
+    exist_rankings = [
+        r for r, in db.query(WorldBossReward.ranking).filter_by(raid_id=raid_id)
+    ]
+    # avatar_address : list of world boss reward amount
+    world_boss_reward_amounts: dict[str, list[dict]] = {}
+    # raid_id,ranking,agent_address,avatar_address,amount,ticker,decimal_places,target_nonce
+    for row in rows:
+        # parse row
+        ranking = int(row[1])
+        agent_address = row[2]
+        avatar_address = row[3]
+        amount = int(row[4])
+        ticker = row[5]
+        decimal_places = int(row[6])
+        nonce = int(row[7])
+
+        # get or create world_boss_reward
+        if ranking not in exist_rankings and not world_boss_rewards.get(avatar_address):
+            world_boss_reward = {
+                "raid_id": raid_id,
+                "ranking": ranking,
+                "agent_address": agent_address,
+                "avatar_address": avatar_address,
+            }
+            world_boss_rewards[avatar_address] = world_boss_reward
+
+        # create world_boss_reward_amount
+        world_boss_reward_amount = {
+            "amount": amount,
+            "decimal_places": decimal_places,
+            "ticker": ticker,
+            "tx_id": tx_ids[nonce],
+        }
+        if not world_boss_reward_amounts.get(avatar_address):
+            world_boss_reward_amounts[avatar_address] = []
+        world_boss_reward_amounts[avatar_address].append(world_boss_reward_amount)
+    if world_boss_rewards:
+        db.execute(insert(WorldBossReward), world_boss_rewards.values())
+    result = db.query(WorldBossReward).filter_by(raid_id=raid_id)
+    values = []
+    for reward in result:
+        exist_tickers = [i.ticker for i in reward.amounts]
+        if world_boss_rewards.get(reward.avatar_address):
+            for amounts in world_boss_reward_amounts[reward.avatar_address]:
+                if amounts["ticker"] not in exist_tickers:
```

Comment:
In case we [overwrite? renew?] reward for same ticker, new reward amount will be ignored by this condition.
Is it ok? Or the overwriting case does not appear at all?

Reply:
ignore duplicate ticker is intended